samcli/lib/utils/managed_cloudformation_stack.py

```diff
@@ -0,0 +1,130 @@
+"""
+Bootstrap's user's development environment by creating cloud resources required by SAM CLI
+"""
+
+import logging
+
+import boto3
+
+import click
+
+from botocore.config import Config
+from botocore.exceptions import ClientError, BotoCoreError, NoRegionError, NoCredentialsError, ProfileNotFound
+
+from samcli.commands.bootstrap.exceptions import ManagedStackError
+from samcli.commands.exceptions import UserException, CredentialsError, RegionError
+
+
+SAM_CLI_STACK_PREFIX = "aws-sam-cli-managed-"
+LOG = logging.getLogger(__name__)
+
+
+def manage_stack(profile, region, stack_name, template_body):
+    try:
+        if profile:
+            session = boto3.Session(profile_name=profile, region_name=region if region else None)
+            cloudformation_client = session.client("cloudformation")
```

Comment:
this is a new logic but doesn't not result in any behavior change as the method caller(see bootstrap.py) always pass profile=None (which how it used to be)

Reply:
I think we should not create a new session, the session is handled here

https://github.com/aws/aws-sam-cli/blob/f084cf2d84709e25fe2b7381bdca0d20d452d317/samcli/cli/context.py#L169-L186

so whenever we need to apply any changes to the AWS services session we can do it from this location only, like if we need to use specific services endpoints, so we can change in the default session, and we are sure that it will be applied to all AWS services connections

Reply:
This use case is different. The one you are referring to is for the --profile option where we execute a SAM CLI command against **one** AWS account. But for this use case we want to deploy a CFN template to **multiple** AWS accounts so we don't manage the AWS account through this --profile option. 

Also we don't